File: src/Dart/fizzbuzz.dart
import 'dart:core';
import 'dart:io';

bool check_factor(int n, int i) {

  if (n % i == 0) {
        
        return true;
  }
  else {
        return false;
  }
}

void main() {
    stdout.write("Input a integer > ");
    String str = stdin.readLineSync().toString();
    int iNumber = int.parse(str);

    for (int i = 1; i <= iNumber; i++) {
      
      if (check_factor(i,3) && check_factor(i,5)) {
        print("FizzBuzz");
      }
      else if (check_factor(i,3)) {
        print("Fizz");
      }
      else if (check_factor(i,5)) {
        print("Buzz");
      }
      else if (check_factor(i,11)) {
        print("Yes");
      }
      else {
        print(i);
      }
    }
}

File: src/Rust/fizzbuzz.rs
use std::io;
use std::io::Write;

fn main() {
    let mut number = String::new();
    print!("Select a number from 1-65,535 > ");
    io::stdout().flush().unwrap();
    io::stdin().read_line(&mut number).expect("Couldn't read line");
    let number: u16 = number.trim().parse().expect("Input a valid number");

    for i in 1..=number {
        if i % 3 == 0 && i % 5 == 0 {
            println!("fizzbuzz");
        } else if i % 3 == 0 {
            println!("fizz");
        } else if i % 5 == 0 {
            println!("buzz");
        } else {
            println!("{i}");
        }
    }
}
